unitTests/ui/MissingTermsTest.ts

```diff
@@ -0,0 +1,192 @@
+          test = mockComponent({ clickable: true });
+          clickOnMissingTerms();
+          expect(missingTermsClickableSpy).toHaveBeenCalled();
+        });
+
+        it('false should not allow the user to click on the missingTerm', () => {
+          test = mockComponent({ clickable: false });
+          clickOnMissingTerms();
+          expect(missingTermsClickableSpy).not.toHaveBeenCalled();
+        });
+      });
+
+      it('caption allows the user to change the title for the missing terms', () => {
+        const caption = 'The missing term';
+        test = mockComponent({ caption });
+        const missingTermsElement = $$(test.cmp.element).find('.coveo-field-caption');
+        expect(missingTermsElement.innerText).toBe(caption);
+      });
+    });
+    describe('when the langage is English', () => {
+      beforeEach(() => {
+        test = mockComponent();
+      });
+      describe('when fetching the missing terms from a query', () => {
+        describe('return', () => {
+          let expectedResult: string[];
+          beforeEach(() => {
+            expectedResult = [...fakeResult.absentTerms];
+          });
+          it('all the missing term present in the query', () => {
+            fakeResult.absentTerms.push(termIsSubWord);
+            expectedResult.push(termIsSubWord);
+            expect(test.cmp.missingTerms.toString()).toBe(expectedResult.toString());
+          });
+
+          it('only the missing term present in the query', () => {
+            fakeResult.absentTerms.push(termNotInQuery);
+            expect(test.cmp.missingTerms.toString()).toBe(expectedResult.toString());
+          });
+        });
+      });
+
+      describe('when re-injecting a term as an exact match', () => {
+        it('and the term is present in the query, the term is re-injected as an exact match', () => {
+          test.cmp.includeTermInQuery(termPresent);
+          expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
+            'q',
+            'This is and "will" be a really really really-really long query!!!'
+          );
+        });
+
+        it('and the term is the first word in the query, the term is re-injected as an exact match', () => {
+          const firstWord = 'This';
+          fakeResult.absentTerms.push(firstWord);
+          test.cmp.includeTermInQuery(firstWord);
+          expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
+            'q',
+            '"This" is and will be a really really really-really long query!!!'
+          );
+        });
+
+        it('and the term is the last word in the query, the term is re-injected as an exact match', () => {
+          const lastWord = 'query';
+          fakeResult.absentTerms.push(lastWord);
+          test.cmp.includeTermInQuery(lastWord);
+          expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
+            'q',
+            'This is and will be a really really really-really long "query"!!!'!!!
```

Comment:
Looks like there are some additional exclamation marks at the end of this line?

Reply:
my bad